server/src/main/java/org/opensearch/index/mapper/NumberFieldMapper.java

```diff
@@ -201,18 +202,37 @@ public Float parse(XContentParser parser, boolean coerce) throws IOException {
             }
 
             @Override
-            public Query termQuery(String field, Object value) {
+            public Query termQuery(String field, Object value, boolean hasDocValues, boolean isSearchable) {
                 float v = parse(value, false);
+                if (isSearchable && hasDocValues) {
+                    Query query = HalfFloatPoint.newExactQuery(field, v);
+                    Query dvQuery = SortedNumericDocValuesField.newSlowExactQuery(field, HalfFloatPoint.halfFloatToSortableShort(v));
+                    return new IndexOrDocValuesQuery(query, dvQuery);
+                }
+                if (hasDocValues) {
+                    return SortedNumericDocValuesField.newSlowExactQuery(field, HalfFloatPoint.halfFloatToSortableShort(v));
+                }
                 return HalfFloatPoint.newExactQuery(field, v);
             }
 
             @Override
-            public Query termsQuery(String field, List<Object> values) {
+            public Query termsQuery(String field, List<Object> values, boolean hasDocValues, boolean isSearchable) {
                 float[] v = new float[values.size()];
+                long points[] = new long[v.length];
                 for (int i = 0; i < values.size(); ++i) {
                     v[i] = parse(values.get(i), false);
+                    points[i] = NumericUtils.floatToSortableInt(v[i]);
+                }
+                if (isSearchable && hasDocValues) {
+                    Query query = FloatPoint.newSetQuery(field, v);
+                    Query dvQuery = SortedNumericDocValuesField.newSlowSetQuery(field, points);
+                    return new IndexOrDocValuesQuery(query, dvQuery);
+                }
+                if (hasDocValues) {
+                    return SortedNumericDocValuesField.newSlowSetQuery(field, points);
                 }
-                return HalfFloatPoint.newSetQuery(field, v);
+                return FloatPoint.newSetQuery(field, v);
```

Comment:
Why we are replacing `HalfFloatPoint.newSetQuery` with `FloatPoint.newSetQuery(field, v)` here?